src/Validator.php

```diff
@@ -46,13 +49,27 @@ public function __construct(
 
     /**
      * @param DataSetInterface|mixed|RulesProviderInterface $data
-     * @param iterable<Closure|Closure[]|RuleInterface|RuleInterface[]>|null $rules
+     * @param class-string|iterable<Closure|Closure[]|RuleInterface|RuleInterface[]>|RulesProviderInterface|null $rules
+     * @param int $rulesPropertyVisibility What visibility levels to use when reading rules from the class specified in
+     * `$rules` argument.
+     *
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
      */
-    public function validate(mixed $data, ?iterable $rules = null): Result
-    {
+    public function validate(
```

Comment:
It's not the same as in the interface: https://github.com/yiisoft/validator/blob/master/src/ValidatorInterface.php#L10

Reply:
Good catch! :+1: Then it's better to move it to a property like in `Nested` rule.